server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1050,29 +1051,24 @@ static Set<String> dataStreamsExclusivelyUsingTemplates(final ClusterState state
             .reduce(Sets::union)
             .orElse(Set.of());
 
+        // Filter and sort all composable templates in advance, to speed up template retrieval later on.
+        var templates = state.metadata()
```

Comment:
So this confused me a little bit, both the name and the comment. It wasn't clear to me what is filtered and when we use the word templates, which templates do we mean. So, what if we rename:
- `templateNames` to `requestedTemplateNames`, signalling that these was provided by the caller.
- `templates` to `restTemplateCandidates` or `otherTemplateCandidates`, signalling that these are the rest, so not the ones provided by the user and candidates because `isGlobalAndHasIndexHiddenSetting` is filtering out the non candidates.

Names are up for debate but I hope I made clear the direction I am considering.

Reply:
Good point, I've renamed the `templates` variable and updated the comments. Let me know if this is more clear.